### PR TITLE
Support transpiling to ES Modules

### PR DESCRIPTION
**Pull request by exchange321**
INT-10222

TLDR, this PR contains:
- Add [rollup](https://github.com/rollup/rollup) for transpiling to ES Modules format
- Upgrade TypeScript to 3.9 to improve rollup build performance
- Fix types/implementations affected by TypeScript upgrade
- Fix @types/styled-components types affected by TypeScript upgrade
- Support ES6 imports of CSS/SCSS files
- Convert named importing lodash methods to importing one by one

UI-Components has always been transpiling to CJS, which causes modern module bundlers such as webpack to not be able to remove the unused code. While this is completely fine in SPA and Lapis as they are more likely to use most of the components in UI-Components, this certainly is a major drawback for its use in a frontend microservice architecture as the whole UI-Components needs to be bundled even though only a fraction of it is being used, leading to a total file size the user has to download much larger.

By supporting ESM format, module bundlers can utilize their dead-code elimination feature to remove the code which isn't being used at all, resulting in a smaller total bundle size.

The following is the production bundle analysis results in [tunnel](https://github.com/intellihr/tunnel):

image

image

image

image

image

From the results, we can observe that not only is the UI-Components size smaller, but the bundle has much fewer dependencies of UI-Components too.

Common chunk file size difference:

| Name             | CJS (KB) | ESM (KB) |
|------------------|----------|----------|
| Navigation Store | 629.55   | 45.8     |
| Error Page Store | 629.55   | 45.81    |
| Navigation       | 700.48   | 316.37   |
| Error Page       | 674.89   | 212.96   |
| Toaster          | 644.95   | 159.04   |

**The following MUST be checked prior to approval. If not relevant to your PR, remove the line from your description.**
- [ ]  The `styleguide.config.js` has been updated to show examples/new functionality for the component
- [ ]  Test Coverage for any new or updated functionality
- [x]  New and updated components have been tested locally in lapis and SPA and work as expected
- [x]  This PR has been tagged with PATCH, MINOR, or MAJOR.
- [ ]  The component has data-component-type and data-component-context attributes following the standard
- [ ]  You have requested a cross-team review on this component so everyone knows it exists

### CODE REVIEW

File: src/domain/Paginators/Pagination/pagination.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+// @ts-ignore
```
Comment (andrewhillcode) [inline src/domain/Paginators/Pagination/pagination.ts:1]: what lol

File: package.json [hunk 2]
```diff
@@ -40,7 +40,6 @@
     "@types/react-test-renderer": "16.8.1",
     "@types/react-tooltip": "^3.3.5",
     "@types/react-transition-group": "2.9.1",
-    "@types/styled-components": "^4.1.16",
```
Comment (andrewhillcode) [inline package.json:43]: What's wrong with `@types/styled-components`?

File: src/domain/Paginators/Pagination/pagination.ts (hunk 1 above)
Comment (andrewhillcode) [reply src/domain/Paginators/Pagination/pagination.ts:1]: Oh I see this file is to hack styleguide into showing examples

Comment (exchange321) [reply src/domain/Paginators/Pagination/pagination.ts:1]: Yeah, having an empty ts file causes rollup typescript plugin to throw an error. This is to mute that.

File: package.json (hunk 2 above)
Comment (exchange321) [reply package.json:43]: Ah~ TS 3.9 does not allow | operator in an extend clause like this
```
export type StyledComponentInnerComponent<
    C extends React.ComponentType<any>
> = C extends
    | StyledComponent<infer I, any, any, any>
    | StyledComponent<infer I, any, any>
    ? I
    : C;
```

So I just change it to
```
export type StyledComponentInnerComponent<
    C extends React.ComponentType<any>
> = C extends StyledComponent<infer I, any, any, any> ? I :
    C extends StyledComponent<infer I, any, any> ? I :
    C;
```

^ is fixed in @types/styled-components@5.0.1 I think, but I guess we don't wanna use that
